physicsnemo/datapipes/cae/domino_datapipe.py

```diff
@@ -816,6 +800,74 @@ def process_data(self, data_dict):
 
             return_dict.update(volume_dict)
 
+        # For domain parallelism, shard everything appropriately:
+        if self.config.shard_grid or self.config.shard_points:
+            # Mesh was defined above!
+            output_dict = {}
+
+            # For scattering, we need to know the _global_ index of rank
+            # 0 on this mesh:
+            global_index = dist.get_global_rank(mesh.get_group(), 0)
+
+            for key, value in return_dict.items():
+                grid_placements = (
+                    [
+                        Shard(0),
+                    ]
+                    if self.config.shard_grid
+                    else [
+                        Replicate(),
+                    ]
+                )
+                point_placements = (
+                    [
+                        Shard(0),
+                    ]
+                    if self.config.shard_points
+                    else [
+                        Replicate(),
+                    ]
+                )
+                if key == "volume_min_max":
+                    output_dict[key] = ShardTensor.from_local(
+                        value,
+                        mesh,
+                        [
+                            Replicate(),
+                        ],
+                    )
+                elif key == "surface_min_max":
+                    output_dict[key] = ShardTensor.from_local(
+                        value,
+                        mesh,
+                        [
+                            Replicate(),
+                        ],
+                    )
+                elif not isinstance(value, ShardTensor):
+                    if "grid" in key:
+                        output_dict[key] = scatter_tensor(
+                            value.contiguous(),
+                            global_index,
+                            mesh,
+                            grid_placements,
+                            global_shape=value.shape,
+                            dtype=value.dtype,
+                        )
+                    else:
+                        output_dict[key] = scatter_tensor(
+                            value.contiguous(),
+                            global_index,
+                            mesh,
+                            point_placements,
+                            global_shape=value.shape,
+                            dtype=value.dtype,
+                        )
+                else:
+                    output_dict[key] = value
```

Comment:
The `ruff` formatting kinda butchered this, it's a simple loop:
- volume_min_max and surface_min_max get replicated across GPUs
- grids get scattered according to "shard_grid"
- everything else is point-like and gets scattered according to "shard_points"
- The whole thing only triggers if at least one of grid / points is sharded.